scripts/vcf_to_zarr.py

```diff
@@ -0,0 +1,312 @@
+                        so that everyone can traverse the directories if they also have read permission.
+
+    """
+
+    # Allow read traversal of directories for everyone (i.e. user, group, other)
+    PERM_ALLOW_DIR_TRAVERSE_MASK = 0o111
+
+    output_zip_path = output_zarr_path + ".zip"
+
+    group_to_region = {}
+    if not region or is_str(region):
+        group_to_region[group] = region
+    else:
+        for subgroup_region in region:
+            subgroup = group + "/" + subgroup_region
+            group_to_region[subgroup] = subgroup_region
+
+    for subgroup, subgroup_region in get_iter(group_to_region):
+        try:
+            allel.vcf_to_zarr(
+                input=input_vcf_path,
+                output=output_zarr_path,
+                group=subgroup,
+                region=subgroup_region,
+                compressor=zarr.Blosc(cname=compress_algo, clevel=compress_level, shuffle=compress_shuffle),
+                overwrite=True,
+                tabix=tabix,
+                fields=fields,
+                alt_number=alt_number,
+                chunk_length=chunk_length,
+                chunk_width=chunk_width,
+                log=log
+            )
+        except StopIteration:
+            # As of scikit-allel v1.1.9, allel.vcf_to_zarr will die with StopIteration
+            # error if the vcf filtered for the desired region is empty.
+            # We want to warn the user and quit nicely when that happens.
+            print ("No entries in vcf {} filtered for region '{}'".format(input_vcf_path, subgroup_region))
+
+            # allele.vcf_to_zarr() will create calldata and variants subgroups before it dies of StopIteration.
+            # but create the subgroup again just in case.
+            root_group = zarr.open_group(output_zarr_path, mode='a', path=subgroup)
+
+
+    # As of scikit-allel v1.1.9 and zarr v2.1.4,
+    # files within the Zarr directory are only read-writeable by the user since
+    # they are created from tempfile.NamedTemporaryFiles.
+    # See https://stackoverflow.com/questions/10541760/can-i-set-the-umask-for-tempfile-namedtemporaryfile-in-python
+    # As a workaround, we recursively set explicit permissions on all files
+    # within the Zarr directory.
+    # We always allow read traversal on zarr subdirectories for everyone,
+    # but allow the user to dictate permissions on the files within the zarr.
+    dir_permissions = permissions | PERM_ALLOW_DIR_TRAVERSE_MASK
+    for curr_dir_path, dir_basenames, file_basenames in os.walk(top=output_zarr_path, topdown=True):
+        for dir_basename in dir_basenames:
+            dir_abs_path = os.path.join(curr_dir_path, dir_basename)
+            os.chmod(dir_abs_path, dir_permissions)
+        for file_basename in file_basenames:
+            file_abs_path = os.path.join(curr_dir_path, file_basename)
+            os.chmod(file_abs_path, permissions)
+
```

Comment:
Previous versions of zarr (until at least zarr v2.1.4) set permissions on files underneath a zarr directory to be read-write by user only, regardless of your umask. The newest version of zarr v2.4.0 will obey your umask. The legacy vcf_to_zarr.py script contains code to traverse the directory tree to explicitly set permissions. This code is no longer required.